sdk/src/inputs.rs

```diff
@@ -0,0 +1,28 @@
+use std::io::{self, Cursor};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use crate::{account_info::LightAccountInfo, proof::ProofRpcResult};
+
+pub struct LightInputs {
+    pub proof: Option<ProofRpcResult>,
+    pub accounts: Option<Vec<LightAccountInfo>>,
```

Comment:
imo this should be a different type not `LightAccountInfo` but for example:
```
pub struct CompressedInputAccount {
    // Owner is inferred from program id
    pub lamports: Option<u64>,
    /// Address of input account (the address can change).
    pub address: Option<[u8; 32]>,
    pub data: Option<[u8; 32]>,
    /// Packed Merkle context of input Merkle context.
		pub merkle_context: PackedMerkleContext,
		pub merkle_tree_root_index: u16,
		pub account_index: u8,
}
```

Reply:
What about `LightAccountMeta`? I really want to avoid leaking the "input" and "output" naming into our API and keep naming similar to Solana.

Reply:
* I think `merkle_tree_root_index` needs to be an `Option<u16>`. When creating a new account, we don't update the MT and the RPC result for a new account doesn't contain the MT root.

Reply:
It also feels weird to me to use `PackedMerkleContext` unconditionally. In case of init accounts, we don't get it from the RPC and have to construct it manually. That's fine.

But what looks off while constructing it is necessity of providing `nullifier_queue_pubkey` (even though is not necessary in that case) and `leaf_index` (the leaf doesn't exist yet).

I'm wondering how to solve this. Since the consensus seems to be strongly against enums, I guess the only option is introducing some new structs (yay!) which are like `MerkleTreeContext` `PackedMerkleTreeContext`, but with the mentioned fields as `Option`s.

Reply:
> LightAccountMeta

think fine to optimistically ship this with what you think is ideal naming for now. i like it.

Reply:
> Also
> 
> * The proposed struct is limited just to input accounts. What about initializing new accounts, where we want to pass an address Merkle tree context?
> * When address changes, we also need an address MT context.

yea this is just for the input state; the generic one (which has a field for the immutable input) is on notion

Reply:
Done, please let me know if the current `AccountMeta` -> `AccountInfo` abstraction looks good to you